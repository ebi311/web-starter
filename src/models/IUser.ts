export type Gender = 'man' | 'woman' | 'none';
export default interface IUser {
  familyName: string;
  givenName: string;
  gender: Gender;
}
