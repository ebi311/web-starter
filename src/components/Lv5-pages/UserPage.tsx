import React from 'react';
import { useSelector } from 'react-redux';
import { IState } from '../../models/IState';
import IUser from '../../models/IUser';
import UserTemplate from '../Lv4-templates/UserTemplate';

const UserPage: React.FC = () => {
  const user = useSelector<IState, IUser>(s => s.user);
  return <UserTemplate user={user} />;
};

export default UserPage;
