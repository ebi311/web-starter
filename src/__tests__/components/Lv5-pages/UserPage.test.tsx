import React from 'react';
import UserPage from '../../../components/Lv5-pages/UserPage';
import { shallow } from 'enzyme';
import IUser from '../../../models/IUser';
import { useSelector } from 'react-redux';

jest.mock('react-redux');

describe('UserPage', () => {
  test('success', () => {
    const user: IUser = {
      familyName: 'nanashi',
      givenName: 'gonbe',
      gender: 'man',
    };
    (useSelector as jest.Mock).mockImplementationOnce(fn => fn({ user }));
    const wrapper = shallow(<UserPage />).dive();
    expect(wrapper).toMatchSnapshot();
  });
});
