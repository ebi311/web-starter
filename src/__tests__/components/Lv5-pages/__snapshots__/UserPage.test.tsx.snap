// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserPage success 1`] = `
<styled.div>
  <UserForm
    user={
      Object {
        "familyName": "nanashi",
        "gender": "man",
        "givenName": "gonbe",
      }
    }
  />
</styled.div>
`;
